File: Easy/E884.py
from typing import List


def uncommonFromSentences(s1: str, s2: str) -> List[str]:

    common = []
    x = s1.split()
    y = s2.split()

    xDict = {}
    yDict = {}

    for i in x:
        if (i in xDict):
            xDict[i] += 1
        else:
            xDict[i] = 1

    for j in y:
        if (j in yDict):
            yDict[j] += 1
        else:
            yDict[j] = 1

    for i in xDict : 
        if xDict.get(i) == 1 and yDict.get(i) == None : common.append(i)

    for j in yDict :
        if yDict.get(j) == 1 and xDict.get(j) == None : common.append(j)
    
    return common


prompt = ["this apple is sweet", "this apple is sour"]
print(uncommonFromSentences(prompt[0], prompt[1]))

File: Easy/E884.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <sstream>
using namespace std;


vector<string> uncommonFromSentences(string s1, string s2) {

    vector<string> common;
    stringstream strs(s1);
    string words;

    auto map1 = unordered_map<string, int>{};
    auto map2 = unordered_map<string, int>{};

    while (getline(strs, words, ' ')){
        map1[words]++;  
        cout << words;
    }
    strs.str(string());

    strs(s2);
    while (getline(strs, words, ' ')) map2[words]++;
    strs.str(string());


    for (auto &it : map1) 
        if (map1[it.first] == 1 && map2[it.first] == 0) common.push_back(map1[it.first]);

    for (auto &it : map2) 
        if (map2[it.first] == 1 && map1[it.first] == 0) common.push_back(map2[it.first]);


    return common;
}


int main(){
    string s1 = "this apple is sweet";
    string s2 = "this apple is sour";

    vector<string> vec = uncommonFromSentences(s1, s2);
    for (auto & it : vec) cout << it << ",";
}